storage/replica_test.go

```diff
@@ -3365,3 +3366,199 @@ func TestReplicaDestroy(t *testing.T) {
+		// returned.
+		{lo: indexes[5], hi: indexes[9], maxBytes: 1, expResultCount: 1},
+		// Case 4: hi value is past the last index, should return all available
+		// entries
+		{lo: indexes[5], hi: indexes[9] + 1, expResultCount: 5},
+		// Case 5: all values have been truncated.
+		{lo: indexes[1], hi: indexes[2], expError: raft.ErrCompacted},
+		// Case 6: hi has just been truncated.
+		{lo: indexes[1], hi: indexes[4], expError: raft.ErrCompacted},
+		// Case 7: another case where hi has just been truncated.
+		{lo: indexes[3], hi: indexes[4], expError: raft.ErrCompacted},
+		// Case 8: lo has been truncated and hi is the truncation point.
+		{lo: indexes[4], hi: indexes[5], expError: raft.ErrCompacted},
+		// Case 9: lo has been truncated but hi is available.
+		{lo: indexes[4], hi: indexes[9], expError: raft.ErrCompacted},
+		// Case 10: lo has been truncated and hi is not available.
+		{lo: indexes[4], hi: indexes[9] + 100, expError: raft.ErrCompacted},
+		// Case 11: lo has been truncated but hi is available, and maxBytes is
+		// set low.
+		{lo: indexes[4], hi: indexes[9], maxBytes: 1, expError: raft.ErrCompacted},
+		// Case 12: lo is available but hi isn't.
+		{lo: indexes[5], hi: indexes[9] + 100, expError: raft.ErrUnavailable},
+		// Case 13: both lo and hi are not available.
+		{lo: indexes[9] + 100, hi: indexes[9] + 1000, expError: raft.ErrUnavailable},
+		// Case 14: lo is available, hi is not, but it was cut off by maxBytes.
+		{lo: indexes[5], hi: indexes[9] + 1000, maxBytes: 1, expResultCount: 1},
+	} {
+		ents, err := rng.Entries(tc.lo, tc.hi, tc.maxBytes)
+		if tc.expError == nil && err != nil {
+			t.Errorf("%d: expected no error, got %s", i, err)
+			continue
+		} else if err != tc.expError {
+			t.Errorf("%d: expected error %s, got %s", i, tc.expError, err)
+			continue
+		}
+		if len(ents) != tc.expResultCount {
+			t.Errorf("%d: expected %d entires, got %d", i, tc.expResultCount, len(ents))
+		}
+	}
+
+	// Case 15: Lo must be less than or equal to hi.
+	if _, err := rng.Entries(indexes[9], indexes[5], 0); err == nil {
+		t.Errorf("15: error expected, got none")
+	}
+
+	// Case 16: add a gap to the indexes.
+	if err := engine.MVCCDelete(tc.store.Engine(), nil, keys.RaftLogKey(rangeID, indexes[6]), roachpb.ZeroTimestamp,
+		nil); err != nil {
+		t.Fatal(err)
+	}
+	if _, err := rng.Entries(indexes[5], indexes[9], 0); err == nil {
+		t.Errorf("16: error expected, got none")
+	}
+
+	// Case 17: don't hit the gap due to maxBytes.
+	ents, err := rng.Entries(indexes[5], indexes[9], 1)
+	if err != nil {
+		t.Errorf("17: expected no error, got %s", err)
+	}
+	if len(ents) != 1 {
+		t.Errorf("17: expected 1 entry, got %d", len(ents))
+	}
+
+	// Case 18: don't hit the gap due to truncation.
+	if _, err := rng.Entries(indexes[4], indexes[9], 0); err != raft.ErrCompacted {
+		t.Errorf("18: expected error %s , got %s", raft.ErrCompacted, err)
+	}
+}
+
+// TestEntires verifies that Entries functions correctly.
```

Comment:
s/Ent(ri|ir)es)/Term/g

Or just delete the comment; tests only need doc-style comments when you have something non-trivial to say.